engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/StorageOrchestrator.java

```diff
@@ -194,8 +197,9 @@ public MigrationResponse migrateData(Long srcDataStoreId, List<Long> destDatasto
                 destDatastoreId = orderedDS.get(1);
             }
 
-            if (chosenFileForMigration.getSize() > storageCapacities.get(destDatastoreId).first()) {
-                s_logger.debug("file: " + chosenFileForMigration.getId() + " too large to be migrated to " + destDatastoreId);
+            if (chosenFileForMigration.getPhysicalSize() > storageCapacities.get(destDatastoreId).first()) {
+                s_logger.debug(String.format("%s: %s too large to be migrated to %s", getObjectType(chosenFileForMigration), chosenFileForMigration.getUuid(), destDatastoreId));
+                skipped += 1;
                 continue;
```

Comment:
Does the files migrated have any relation (eg. template info and template file)? Is it ok to migrate other files, if one large file is skipped in a group of related files? or Is it better to check with the cumulative size of all the related files?

Reply:
You are right @sureshanaparti cc @DaanHoogland, we have a utility method getFileSize(), that gets the cumulative size. Since we are on that topic, I do have a doubt now, currently only snapshot chains are migrated together, is there any constraint on templates too, i.e., do multi-disk templates also need to go to the same store?

Reply:
I wouldn't know for sure, but I think not. templates are gotten from their store and copied to primary for usage. no reason for a multi-disk template to be on one store asfaics @Pearl1594 @sureshanaparti .

Reply:
@nvazquez can you comment on this discussion, please?

Reply:
Afaik multi-disk templates wouldn't necessarily need to go to the same store

Reply:
I tried moving one of the child templates of a multi-disk template to another store and modified the store_id details in the template_store_ref table and noticed a failure during deletion of the template, where it complained about not being able to find the template.  cc @DaanHoogland @nvazquez

Reply:
@Pearl1594 I've tested with your last patch and indeed all mulidisk parts go to the same store (migrating a store with four multidisk templates to two destinations, the load gets balanced but templates don't get split)
cc @nvazquez @sureshanaparti